sycl/test/extensions/annotated_usm/annotated_usm.cpp

```diff
@@ -0,0 +1,243 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -fsyntax-only -Xclang -verify -Xclang -verify-ignore-unexpected=note %s
+// expected-no-diagnostics
+
+// Common tests for annotated USM allocation
+
+#include "sycl/sycl.hpp"
+
+#include <iostream>
+
+// clang-format on
+
+using namespace sycl::ext::oneapi::experimental;
+using namespace sycl::ext::intel::experimental;
+using alloc = sycl::usm::alloc;
+
+constexpr int N = 10;
+
+void TestAlloc(sycl::queue &q) {
+  // P1: input properties
+  properties P1{conduit, stable, cache_config{large_slm}};
+  // ANNP1: output properties with usm_kind and no runtime property
+  // `cache_config`
+  properties ANNP1{conduit, stable, usm_kind<alloc::device>};
+  auto APtr1 = malloc_device_annotated<int>(N, q, P1);
+  auto APtr11 = malloc_device_annotated<int, decltype(P1)>(N, q, P1);
+  auto APtr12 =
+      malloc_device_annotated<int, decltype(P1), decltype(ANNP1)>(N, q, P1);
+  // Check that the annotated_ptr allocated should contain properties in ANNP1
+  static_assert(
+      std::is_same_v<decltype(APtr1), annotated_ptr<int, decltype(ANNP1)>>);
+  static_assert(std::is_same_v<decltype(APtr11), decltype(APtr11)>);
```

Comment:
This assert and the next one seem trivially true? Since you are comparing APtr11 with APtr11, of course they have the same type.